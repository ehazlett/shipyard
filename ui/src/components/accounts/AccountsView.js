import React from 'react';

import { Route, Switch, Redirect } from "react-router-dom";

import CreateAccountView from './CreateAccountView';
import AccountListView from './AccountListView';
import AccountInspectView from './AccountInspectView';

class AccountsView extends React.Component {
  render() {
    return (
      <Switch>
        <Route exact path="/accounts" component={AccountListView} />
        <Route exact path="/accounts/inspect/:id" component={AccountInspectView} />
        <Route exact path="/accounts/create" component={CreateAccountView} />
        <Route render={() => <Redirect to="/accounts" />} />
      </Switch>
    );
  }
}

export default AccountsView;
