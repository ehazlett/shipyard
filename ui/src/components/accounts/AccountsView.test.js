import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router";
import AccountsView from "./AccountsView";
import renderer from "react-test-renderer";

it("AccountsView renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <AccountsView />
    </MemoryRouter>,
    div
  );
});

it("AccountsView snapshot matches", () => {
  const tree = renderer
    .create(<MemoryRouter><AccountsView /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
