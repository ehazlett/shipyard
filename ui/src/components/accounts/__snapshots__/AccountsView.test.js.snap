exports[`test AccountsView snapshot matches 1`] = `
<div
  className="ui active transition visible inverted dimmer"
  onClick={[Function]}>
  <div
    className="content">
    <div
      className="center">
      <div
        className="ui loader" />
    </div>
  </div>
</div>
`;
